src/lib/core/backtrace.cc

```diff
@@ -246,11 +253,18 @@ __asm__ volatile(
 	"\tmovq 40(%%rsp), %%rbp\n"
 	/* Set first arg and call */
 	"\tmovq %0, %%rdi\n"
-#ifdef TARGET_OS_DARWIN
-	"\tandq $0xfffffffffffffff0, %%rsp\n"
-#endif
+#ifdef HAVE_CFI_ASM
+	".cfi_remember_state\n"
+	".cfi_def_cfa %%rsp, 8 * 7\n"
+	"\tleaq %P2(%%rip), %%rax\n"
+	"\tcall *%%rax\n"
+	".cfi_restore_state\n"
+#else
+#warning "no compiler support for CFI assembly: backtracing from a suspended"
+	 "libcoro fiber may be broken";
```

Comment:
As I mentioned in the commit message:

> We need to know the exact location of the stack pointer: replace the
16-byte stack alignment instruction on x86_64 macOS by adding the
`force_align_arg_pointer` attribute to `unw_getcontext_f`.

Reply:
> Instead you just remove it - it boggles me.

I did not just remove it — I replaced it with an equivalent annotation for the function which we set up the call for:
https://github.com/tarantool/tarantool/pull/6889/commits/8c269f183bf120dbc22f012c72e77d90e26abeb3#diff-0bec7f0a8f47c6ef5b405691cc66deb34893e32159f373666e87c1b12b2f2fabR199-R205. Instead of doing this alignment explicitly, the compiler will do it implicitly in the called function's prologue.

See _force_align_arg_pointer_ attribute, for instance: https://clang.llvm.org/docs/AttributeReference.html#force-align-arg-pointer.

Reply:
> If I remember correct, we discussed this issue earlier and came to a conclusion the align should be platform-agnostic.

I had a discussion about this matter with @alyapunov, but without regard that we had this stack alignment code only for macOS (macOS syscalls rely on 16-byte stack alignment and therefore it is critical to adhere to the ABI on this specific platform) — I do not see any reason why we should start aligning the stack for other platforms too, if it is not strictly required and it worked to this day.

Not to mention that we have an x86_64 CentOS runner with an ancient GCC version (4.8) shipped by default, which does not support the aforementioned _force_align_arg_pointer_ attribute — mandating stack alignment here would mean giving up backtraces for this configuration and adding another compiler feature dependency (assuming macOS ships bleeding-edge clang versions that support all features reliably).

Reply:
But I guess we cannot be sure this code is correct, and this is probably UB (given that we make library calls lower further on).

Reply:
> > Instead you just remove it - it boggles me.
> 
> I did not just remove it — I replaced it with an equivalent annotation for the function which we set up the call for: [8c269f1#diff-0bec7f0a8f47c6ef5b405691cc66deb34893e32159f373666e87c1b12b2f2fabR199-R205](https://github.com/tarantool/tarantool/commit/8c269f183bf120dbc22f012c72e77d90e26abeb3#diff-0bec7f0a8f47c6ef5b405691cc66deb34893e32159f373666e87c1b12b2f2fabR199-R205). Instead of doing this alignment explicitly, the compiler will do it implicitly in the called function's prologue.
> 
> See _force_align_arg_pointer_ attribute: [clang](https://clang.llvm.org/docs/AttributeReference.html#force-align-arg-pointer) or [GCC](https://gcc.gnu.org/onlinedocs/gcc/x86-Function-Attributes.html#x86-Function-Attributes) references.

Compiler-generated alignment will happen only in case it supports force_align attribute. For the case it doesn't - you have nothing to align the stack, just issue a warning. I believe align of the stack hurts no more than use up to 15 bytes more. The reason to do it on other platforms - switch to a 16-byte alignment happens at some time in the past, one can happen to have a mix of libraries, both 8 and 16 aligned. Call to a former from the latter will not cause a problem, while the opposite _can_ cause.

Note, that for 16-byte aligned stack the `AND` in question is effectively a `NOP`.

Reply:
Fixed:

* enabled 16-byte alignment fix for all platforms
* moved this compiler feature check to CMake in the scope of #6877 (will need to merge it first): https://github.com/tarantool/tarantool/pull/6877/files#diff-abe123d6b477234b7b16bf1b8d33438cc8e62bafaed6e92b288f319aca3adaceR146-R154

Reply:
great!